src/main/kotlin/com/openlattice/organizations/DatasetController.kt

```diff
@@ -119,36 +122,61 @@ class DatasetController : DatasetApi, AuthorizingComponent {
             @PathVariable(ID) organizationId: UUID,
             @PathVariable(TABLE_NAME) tableName: String,
             @PathVariable(COLUMN_NAME) columnName: String): OrganizationExternalDatabaseColumn {
-        val tableId = getExternalDatabaseObjectId(organizationId, tableName)
-        val columnId = getExternalDatabaseObjectId(tableId, columnName)
-        ensureReadAccess(AclKey(organizationId, tableId, columnId))
-        return edms.getOrganizationExternalDatabaseColumn(columnId)
+        val tableFqnToId = getExternalDatabaseObjectFqnToIdPair(organizationId, tableName)
+        val columnFqnToId = getExternalDatabaseObjectFqnToIdPair(tableFqnToId.second, columnName)
+        ensureReadAccess(AclKey(tableFqnToId.second, columnFqnToId.second))
+        return edms.getOrganizationExternalDatabaseColumn(columnFqnToId.second)
+    }
+
+    //TODO Metadata update probably won't work
+    @Timed
+    @PatchMapping(path = [ID_PATH + TABLE_NAME_PATH + EXTERNAL_DATABASE_TABLE])
+    override fun updateExternalDatabaseTable(
+            @PathVariable(ID) organizationId: UUID,
+            @PathVariable(TABLE_NAME) tableName: String,
+            @RequestBody metadataUpdate: MetadataUpdate) {
+        val tableFqnToId = getExternalDatabaseObjectFqnToIdPair(organizationId, tableName)
+        ensureAdminAccess()
+        edms.updateOrganizationExternalDatabaseTable(organizationId, tableFqnToId, metadataUpdate)
+    }
+
+    @Timed
+    @PatchMapping(path = [ID_PATH + TABLE_NAME_PATH + COLUMN_NAME_PATH + EXTERNAL_DATABASE_COLUMN])
+    override fun updateExternalDatabaseColumn(
+            @PathVariable(ID) organizationId: UUID,
+            @PathVariable(TABLE_NAME) tableName: String,
+            @PathVariable(COLUMN_NAME) columnName: String,
+            @RequestBody metadataUpdate: MetadataUpdate) {
+        val tableFqnToId = getExternalDatabaseObjectFqnToIdPair(organizationId, tableName)
+        val columnFqnToId = getExternalDatabaseObjectFqnToIdPair(tableFqnToId.second, columnName)
+        ensureAdminAccess()
+        edms.updateOrganizationExternalDatabaseColumn(organizationId, tableFqnToId, columnFqnToId, metadataUpdate)
     }
 
     @Timed
     @DeleteMapping(path = [ID_PATH + TABLE_NAME_PATH + EXTERNAL_DATABASE_TABLE])
     override fun deleteExternalDatabaseTable(
             @PathVariable(ID) organizationId: UUID,
             @PathVariable(TABLE_NAME) tableName: String) {
-        val tableId = getExternalDatabaseObjectId(organizationId, tableName)
-        val aclKey = AclKey(organizationId, tableId)
+        val tableFqnToId = getExternalDatabaseObjectFqnToIdPair(organizationId, tableName)
```

Comment:
I would do the same here as in the later comment: https://github.com/openlattice/datastore/pull/481/files#r380871393

You could just call: `deleteExternalDatabaseTables(organizationId, setOf(tableName))`

Reply:
So, also here I would just call the function in this class:
```
@Timed
    @DeleteMapping(path = [ID_PATH + TABLE_NAME_PATH + EXTERNAL_DATABASE_TABLE])
    override fun deleteExternalDatabaseTable(
            @PathVariable(ID) organizationId: UUID,
            @PathVariable(TABLE_NAME) tableName: String) {
    deleteExternalDatabaseTables(organizationId, setOf(tableName))
}
```